src/label_view.js

```diff
@@ -1,145 +1,213 @@
+    $button.attr('class', "follow-unfollow open-iframe");
+    $button.attr('style', "width: 90%");
 
-        return $parent_div
-    }
+    $preview = $('<div>').html("Preview");
+    $button.append($preview);
+    
+    $bclvframe = $('<div>');
+    $bclvframe.attr('class', "bclv-frame").attr('id', id);
 
-    // iterate over page to get album IDs and append buttons with value
-    $('li[data-item-id]').each(function(index, item){
-        var id = $(item).closest('li').attr('data-item-id');
-        if (id.split("-")[0] == "album"){
-            id = id.split("-")[1];
-        }
+    // add checkbox with stores history of clicks
+    $checkbox = $('<button>');
+    $checkbox.attr('style', "margin: 0px 0px 0px 5px; width: 28px; height: 28px; position: absolute;");
+    
+    toggleState = false
+    if(storageCache.includes(id))
+    {
+        console.log("id exists: ", id);
+        toggleState = true;
+    }
 
-        $preview_element = generatePreview(id);
-        $(item).append($preview_element);
-    });
+    $checkbox = toggleHistorybox($checkbox, toggleState);
 
-    // iterate over page to get album IDs and append buttons with value
-    $('li[data-tralbumid][data-tralbumtype="a"]').each(function(index, item){
-        var id = $(item).attr('data-tralbumid');
-        
-        $preview_element = generatePreview(id);
-        $(item).append($preview_element);
-    });
+    $parent_div.append($button);
+    $parent_div.append($checkbox);
+    $parent_div.append($bclvframe);
 
-    $('.open-iframe').on('click', 
-        function(event){
-            fillframe(event);
-        }
-    );
+    return $parent_div
+}
 
-    $('.historybox').on('click', 
-        function(event){
-            boxclick(event);
-        }
-    );
+chrome.extension.sendMessage({}, function(response) {
 
- }
- }, 10);
+    $( document ).ready(function() {
+        storageLoadedPromise.then(function() {
+            // iterate over page to get album IDs and append buttons with value
+            $('li[data-item-id]').each(function(index, item){
+                var id = $(item).closest('li').attr('data-item-id');
+                if (id.split("-")[0] == "album" || id.split("-")[0] == "track"){
+                    id = id.split("-")[1];
+                }
+
+                $preview_element = generatePreview(id);
+                $(item).append($preview_element);
+            });
+
+            $('li[data-tralbumid][data-tralbumtype="a"]').each(function(index, item){
+                var id = $(item).attr('data-tralbumid');
+                
+                $preview_element = generatePreview(id);
+                $(item).append($preview_element);
+            });
+
+            // catched ID album pages
+            $('#pagedata').first().each(function(index, item){
+                var datablob = JSON.parse($(item).attr("data-blob"));
+                try {
+                    var id = datablob.lo_querystr.split("item_id=")[1].split("&item_type=")[0]
```

Comment:
This is effective, but since this is a URL using [URLSearchParams](https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams) might work:

```js
var urlParams = new URLSearchParams(datablob.lo_querystr);
var id = urlParams.get('item_id')
```

Seems to be supported in [Chrome since 2016 or so](https://caniuse.com/#search=URLSearchParams).

Reply:
works. nice!
updated.